ara/exec/parser/manifest_parser.cpp

```diff
@@ -1,23 +1,122 @@
 #include "manifest_parser.h"
-
+#include <fstream>
+#include <iostream>
 #include <nlohmann/json.hpp>
+#include <stdexcept>
+#include <vector>
 
 namespace ara {
 namespace exec {
 namespace parser {
 
-ExecutionManifest parse_execution_manifest(const std::string& /*path*/)
+using json = nlohmann::json;
+
+namespace {
+namespace EMJsonKeys {
+const std::string kApplicationManifest = "Application_manifest";
+const std::string kApplicationManifestId = "Application_manifest_id";
+// const std::string kProcess = "Process";
+// const std::string kProcessName = "Process_name";
+// const std::string kModeDependentStartupConfigs = "Mode_dependent_startup_configs";
+// const std::string kStartupOptions = "Startup_options";
+// const std::string kStartupOptionsOptionKind = "Option_kind";
+// const std::string kStartupOptionsOptionName = "Option_name";
+// const std::string kStartupOptionsOptionArg = "Option_arg";
+// const std::string kModeInMachineInstanceRefs = "Mode_in_machine_instance_refs";
+// const std::string kFunctionGroup = "Function_group";
+// const std::string kMode = "Mode";
+// TODO add rest of the keys
+
+const std::vector<std::string> kAsVector{kApplicationManifest,
+                                         kApplicationManifestId /* TODO add rest of the keys */};
+
+}  // namespace EMJsonKeys
+
+namespace MMJsonKeys {
+const std::string kMachineManifest = "Machine_manifest";
+// TODO add rest of the keys
+
+const std::vector<std::string> kAsVector{kMachineManifest /* TODO add rest of the keys */};
+}  // namespace MMJsonKeys
+
+}  // anonymous namespace
+
+template <typename T>
+bool try_to_read_value(const json& json_obj, const std::string& key, T& output_value)
+{
+    T tmp{};
+    try {
+        tmp = json_obj.at(key);
+    } catch (json::out_of_range) {
+        std::cerr << "ara::exec::parser::try_to_read_value -> unable to read key: " << key << '\n';
+        return false;
+    }
+    output_value = tmp;
+    return true;
+}
+
+bool validate_manifest(const json& /* json_obj */, const std::vector<std::string>& /* keys */)
+{
+    // TODO implement this method
+    return true;
+}
+
+ExecutionManifest parse_execution_manifest(const json& /* json_obj */)
 {
     // TODO implement this method
     return ExecutionManifest();
 }
 
-MachineManifest parse_machine_manifest(const std::string& /*path*/)
+MachineManifest parse_machine_manifest(const json& /* json_obj */)
 {
     // TODO implement this method
     return MachineManifest();
 }
 
+Manifest parse_manifest(const std::string& path) noexcept(false)
```

Comment:
See my comment in header file.

Reply:
Thanks, I will remove it from implementation. I would rather keep it in header